contracts/src/utils/airdrop/merkle/MerkleAirdrop.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+// interfaces
+import {IMerkleAirdrop} from "./IMerkleAirdrop.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+// libraries
+import {ECDSA} from "solady/utils/ECDSA.sol";
+import {MerkleProofLib} from "solady/utils/MerkleProofLib.sol";
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+
+// contracts
+import {EIP712Base} from "contracts/src/diamond/utils/cryptography/signature/EIP712Base.sol";
+import {CustomRevert} from "contracts/src/utils/libraries/CustomRevert.sol";
+import {Facet} from "contracts/src/diamond/facets/Facet.sol";
+
+contract MerkleAirdrop is IMerkleAirdrop, EIP712Base, Facet {
+  // keccak256("AirdropClaim(address account,uint256 amount)");
+  bytes32 private constant MESSAGE_TYPEHASH =
+    0xaa726e564e52b64144617a6a46c42e8b763d4d224ca1a3e13c1491f8a4763a23;
+
+  function __MerkleAirdrop_init(
+    bytes32 merkleRoot,
+    IERC20 token
+  ) external initializer {
+    _addInterface(type(IMerkleAirdrop).interfaceId);
+    __MerkleAirdrop_init_unchained(merkleRoot, token);
+  }
+
+  /// @inheritdoc IMerkleAirdrop
+  function getMerkleRoot() public view returns (bytes32) {
+    return MerkleAirdropStorage.layout().merkleRoot;
+  }
+
+  /// @inheritdoc IMerkleAirdrop
+  function getToken() public view returns (IERC20) {
+    return MerkleAirdropStorage.layout().token;
+  }
+
+  /// @inheritdoc IMerkleAirdrop
+  function getMessageHash(
+    address account,
+    uint256 amount
+  ) public view returns (bytes32) {
+    return
+      _hashTypedDataV4(
+        keccak256(abi.encode(MESSAGE_TYPEHASH, AirdropClaim(account, amount)))
+      );
+  }
+
+  /// @inheritdoc IMerkleAirdrop
+  function claim(
+    address account,
+    uint256 amount,
+    bytes32[] calldata merkleProof,
+    bytes calldata signature
```

Comment:
```suggestion
    bytes32[] calldata merkleProof
```
The function can be permissionless because the token is only transferred to an `account` included in the tree.

Reply:
with operators we can have them sign but with optionality to send to a separate address

Reply:
Then a `recipient` argument is needed.